packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -0,0 +1,489 @@
+        const tokenIndexAfter = await (positionManager as INonfungiblePositionManager).totalSupply();
+        const tokenId = await (positionManager as INonfungiblePositionManager).tokenByIndex(tokenIndexAfter.sub(1));
+        const position = await (positionManager as INonfungiblePositionManager).positions(tokenId)
+
+        expect(BigNumber.from(vaultAfter.NftCollateralId).eq(tokenId)).to.be.true;
+        expect(position.tickLower === -887220).to.be.true
+        expect(position.tickUpper === 0).to.be.true
+        expect(vaultAfter.shortAmount.sub(vaultBefore.shortAmount).eq(mintWSqueethAmount)).to.be.true
+        expect(vaultAfter.collateralAmount.eq(vaultBefore.collateralAmount)).to.be.true
+      })
+    })
+
+    it("open short in new vault, mint, LP oSQTH only, deposit LP NFT and withdraw ETH collateral", async () => {
+      const vaultId = (await shortSqueeth.nextId());
+      const normFactor = await controller.getExpectedNormalizationFactor()
+      const mintWSqueethAmount = ethers.utils.parseUnits('40')
+      const mintRSqueethAmount = mintWSqueethAmount.mul(normFactor).div(one)
+      const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+      const scaledEthPrice = ethPrice.div(10000)
+      const debtInEth = mintRSqueethAmount.mul(scaledEthPrice).div(one)
+      const collateralToMint = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+      const collateralToFlashloan = collateralToMint.div(2)
+      const flashloanWMintDepositNftParams = {
+        vaultId: 0,
+        wPowerPerpAmount: mintWSqueethAmount.toString(),
+        collateralToDeposit: collateralToMint.toString(),
+        collateralToFlashloan: collateralToFlashloan.toString(),
+        collateralToLp: BigNumber.from(0),
+        collateralToWithdraw: 0,
+        lpAmount0Min: 0,
+        lpAmount1Min: 0,
+        lpLowerTick: -887220,
+        lpUpperTick: 0
+      }
+
+      await controllerHelper.connect(depositor).flashloanWMintDepositNft(flashloanWMintDepositNftParams, {value: collateralToMint.div(2).add(ethers.utils.parseUnits('0.01'))})
+
+      const vaultAfter = await controller.vaults(vaultId)
+      const tokenIndexAfter = await (positionManager as INonfungiblePositionManager).totalSupply();
+      const tokenId = await (positionManager as INonfungiblePositionManager).tokenByIndex(tokenIndexAfter.sub(1));
+      const position = await (positionManager as INonfungiblePositionManager).positions(tokenId)
+
+      expect(BigNumber.from(vaultAfter.NftCollateralId).eq(tokenId)).to.be.true;
+      expect(position.tickLower === -887220).to.be.true
+      expect(position.tickUpper === 0).to.be.true
+      expect(vaultAfter.shortAmount.eq(mintWSqueethAmount)).to.be.true
+      expect(vaultAfter.collateralAmount.eq(collateralToMint.sub(collateralToFlashloan))).to.be.true
+    })
+
+    it("open short, mint with >0 ETH collateral, LP oSQTH + ETH, deposit LP NFT and withdraw ETH collateral", async ()=> {
```

Comment:
no collateral is withdrawn here right?

maybe with these you mean flashloan repaid (but confused me if so)

Reply:
yes meant flashloan repaid, yeah I guess it is confusing with the `collateralToWithdraw` param, will rename the tests